### DISCUSSION TITLE
Exportable class for web components?

### DISCUSSION THREAD

**Discussion by cubap** [digirati-co-uk/iiif-canvas-panel#269] [Q&A] [answered]
I'm finally experimenting with canvas-panel and related bits. All the straightforward examples work easily! I'm interesting in registering a different name to an element and embed a canvas-panel inside another component. In the past, I've done this by importing the component class and using `customElements.define()` but I'm not seeing any obviously named exports that I can use. Am I missing an obvious pattern?

**Comment by stephenwf** [comment, marked as answer]
I think it would require a quick code change:
https://github.com/digirati-co-uk/iiif-canvas-panel/blob/main/packages/canvas-panel/src/index.ts#L1-L6

Changing these from `import` to `export * from` should allow you to grab the class from the module. Theres a PR up #237 which has quite a few dependency changes, would be a good fit there.
